src/usr/local/buildpack/utils/filesystem.sh

```diff
@@ -25,6 +25,8 @@ function find_versioned_tool_path () {
 
 function create_versioned_tool_path () {
   install_dir=$(get_install_dir)
+  # shellcheck disable=SC2174
+  mkdir -p -m 775 "${install_dir}/${TOOL_NAME}"
   mkdir -p "${install_dir}/${TOOL_NAME}/${TOOL_VERSION}"
```

Comment:
This needs to be 755 when not running as root too.

Reply:
We can change the global umasks to `755` and `775` and use the refactored methods.
How important is `read+execute` for `other` for you?

If not at all, then we can use the current umasks and simply replace the function.

Reply:
I think 770 and 750 are ok.

Reply:
I pushed with 775 and 755 first. Lets keep the legacy install compatible and us 770 and 750 for the new tools going forward.